src/server/game/Entities/Player/Player.cpp

```diff
@@ -16460,7 +16460,7 @@ bool Player::HasQuestForItem(uint32 itemid) const
             // This part for ReqItem drop
             for (uint8 j = 0; j < QUEST_ITEM_OBJECTIVES_COUNT; ++j)
             {
-                if (itemid == qinfo->RequiredItemId[j] && q_status.ItemCount[j] < qinfo->RequiredItemCount[j])
+                if ((itemid == qinfo->RequiredItemId[j] && q_status.ItemCount[j] < qinfo->RequiredItemCount[j]) || (turnIn && itemid == qinfo->RequiredItemId[j] && q_status.ItemCount[j] <= qinfo->RequiredItemCount[j]))
```

Comment:
Yea, I saw this after I already committed the removal of the other redundant check, but I had already logged out for the day. As soon as I can get back on my dev box I was going to push it. But again, thanks.

The <= is needed because this check would fail if (for example) you had 2 quests that needed item 1234 and you had 4 of that item. The first quest needs 2, and the second quest needs 2. If you just check for < it would fail because you have the correct amount for the second quest. So on turnin, all of item 1234 would be deleted and you are up the creek trying to turning the second quest. There is never a situation where you would be turning in a quest and need to check for just < and of course we don't care about > :)